autofs-5.1.6 - move AUTOFS_LIB to end of build rule lines

From: Ian Kent <raven@themaw.net>

Move the macro for the autofs.a library to the end of build rule
lines in preparation to converting autofs.a to a shared library.

This results in no functional change.

Signed-off-by: Ian Kent <raven@themaw.net>
---
 CHANGELOG        |    1 +
 Makefile.rules   |    2 +-
 daemon/Makefile  |    2 +-
 modules/Makefile |   12 ++++++------
 4 files changed, 9 insertions(+), 8 deletions(-)

--- autofs-5.1.6.orig/CHANGELOG	2020-12-12 17:59:40.137000000 +0800
+++ autofs-5.1.6/CHANGELOG	2020-12-12 18:25:57.882000000 +0800
@@ -10,6 +10,7 @@
 - fix trailing dollar sun entry expansion.
 - initialize struct addrinfo for getaddrinfo() calls.
 - fix quoted string length calc in expandsunent().
+- move AUTOFS_LIB to end of build rule lines.
 
 
 07/10/2019 autofs-5.1.6
--- autofs-5.1.6.orig/Makefile.rules	2020-12-12 17:59:35.343000000 +0800
+++ autofs-5.1.6/Makefile.rules	2020-12-12 18:25:57.882000000 +0800
@@ -71,5 +71,5 @@
 	$(CC) $(CFLAGS) -S $<
 
 .c.so:
-	$(CC) $(SOLDFLAGS) $(CFLAGS) -o $*.so $< $(LDFLAGS) $(AUTOFS_LIB) $(LIBS)
+	$(CC) $(SOLDFLAGS) $(CFLAGS) -o $*.so $< $(LDFLAGS) $(LIBS) $(AUTOFS_LIB)
 	$(STRIP) $*.so
--- autofs-5.1.6.orig/daemon/Makefile	2020-12-12 17:59:35.343000000 +0800
+++ autofs-5.1.6/daemon/Makefile	2020-12-12 18:25:57.882000000 +0800
@@ -35,7 +35,7 @@
 all: automount
 
 automount: $(OBJS) $(AUTOFS_LIB)
-	$(CC) $(DAEMON_LDFLAGS) -o automount $(OBJS) $(LDFLAGS) $(AUTOFS_LIB) $(LIBS)
+	$(CC) $(DAEMON_LDFLAGS) -o automount $(OBJS) $(LDFLAGS) $(LIBS) $(AUTOFS_LIB)
 	$(STRIP) automount
 
 clean:
--- autofs-5.1.6.orig/modules/Makefile	2020-12-12 17:59:35.343000000 +0800
+++ autofs-5.1.6/modules/Makefile	2020-12-12 18:25:57.882000000 +0800
@@ -110,7 +110,7 @@
 
 parse_amd.so: parse_amd.c amd_parse.tab.o amd_tok.o
 	$(CC) $(SOLDFLAGS) $(CFLAGS) -o parse_amd.so \
-		parse_amd.c amd_parse.tab.o amd_tok.o $(LDFLAGS) $(AUTOFS_LIB) $(LIBS)
+		parse_amd.c amd_parse.tab.o amd_tok.o $(LDFLAGS) $(LIBS) $(AUTOFS_LIB)
 	$(STRIP) parse_amd.so
 
 #
@@ -118,17 +118,17 @@
 #
 lookup_yp.so: lookup_yp.c
 	$(CC) $(SOLDFLAGS) $(CFLAGS) $(NSLCFLAGS) -o lookup_yp.so \
-		lookup_yp.c $(LDFLAGS) $(AUTOFS_LIB) $(LIBS) $(NSLLIB)
+		lookup_yp.c $(LDFLAGS) $(LIBS) $(NSLLIB) $(AUTOFS_LIB)
 	$(STRIP) lookup_yp.so
 
 lookup_nisplus.so: lookup_nisplus.c
 	$(CC) $(SOLDFLAGS) $(CFLAGS) $(NSLCFLAGS) -o lookup_nisplus.so \
-		lookup_nisplus.c $(LDFLAGS) $(AUTOFS_LIB) $(LIBS) $(NSLLIB)
+		lookup_nisplus.c $(LDFLAGS) $(LIBS) $(NSLLIB) $(AUTOFS_LIB)
 	$(STRIP) lookup_nisplus.so
 
 lookup_hesiod.so: lookup_hesiod.c
 	$(CC) $(SOLDFLAGS) $(CFLAGS) $(HESIOD_FLAGS) -o lookup_hesiod.so \
-		lookup_hesiod.c $(LDFLAGS) $(AUTOFS_LIB) $(LIBHESIOD) $(LIBRESOLV) $(LIBS)
+		lookup_hesiod.c $(LDFLAGS) $(LIBHESIOD) $(LIBRESOLV) $(LIBS) $(AUTOFS_LIB)
 	$(STRIP) lookup_hesiod.so
 
 cyrus-sasl.o: cyrus-sasl.c
@@ -140,11 +140,11 @@
 lookup_ldap.so: lookup_ldap.c dclist.o base64.o $(SASL_OBJ)
 	$(CC) $(SOLDFLAGS) $(CFLAGS) $(LDAP_FLAGS) -o lookup_ldap.so \
 		lookup_ldap.c dclist.o base64.o $(SASL_OBJ) \
-		$(LDFLAGS) $(AUTOFS_LIB) $(LIBLDAP) $(LIBRESOLV) $(LIBS)
+		$(LDFLAGS) $(LIBLDAP) $(LIBRESOLV) $(LIBS) $(AUTOFS_LIB)
 	$(STRIP) lookup_ldap.so
 
 mount_nfs.so: mount_nfs.c replicated.o
 	$(CC) $(SOLDFLAGS) $(CFLAGS) -o mount_nfs.so \
-		mount_nfs.c replicated.o $(LDFLAGS) $(AUTOFS_LIB) $(LIBS)
+		mount_nfs.c replicated.o $(LDFLAGS) $(LIBS) $(AUTOFS_LIB)
 	$(STRIP) mount_nfs.so
 
